specs/angular-readonly-form-styling/README.md

```diff
@@ -0,0 +1,118 @@
+
+## Links To Relevant Work Items and Reference Material
+
+[SLE control pattern docs](https://stratus-storybook.ni.dev/?path=/docs/patterns-control-text--docs)
+
+## Implementation / Design
+
+To enable this new design, the following changes need to be made within the nimble repo, each of which is discussed in more detail below:
+
+- Introduce `readonly-disabled-appearance` configuration on the `nimble-text-field`, `nimble-number-field`, `nimble-text-area`, `nimble-select`, and `nimble-combobox`
+- Introduce `frameless` appearance to the `nimble-select`, `nimble-combobox`, and `nimble-number-field`
+- Introduce `full-bleed` configuration on the `nimble-select`, `nimble-combobox`, and `nimble-number-field`
+
+### Read-only appearance when disabled
+
+The `nimble-text-field`, `nimble-number-field`, `nimble-text-area`, `nimble-select`, and `nimble-combobox` will be updated to have a new boolean attribute named `readonly-disabled-appearance`, which will be associated with a new `readOnlyDisabledAppearance` boolean property.
+
+When these components have this property set to `true` while being disabled, the component will remain disabled from an ARIA and focusability point of view, but it will be styled as a read-only control. This includes:
+
+- Not dimming the control label
+- Not dimming the control value
+- [select] Hiding drop-down arrow
+- [combobox] Hiding drop-down button
+
+The `readOnlyDisabledAppearance` property will have no effect on a component that is not disabled.
+
+### Frameless select, combobox, and number field
+
+The visual design for read-only controls in forms is for them to be rendered without any borders or backgrounds. This matches the `frameless` appearance nimble has today for the `nimble-text-field`. Therefore, the select, combobox, and number field will all be updated to also support a `frameless` appearance.
+
+### Full-bleed select, combobox, and number field
+
+The visual design for read-only controls in forms is for them to be rendered without any left padding on the value (i.e. the value is aligned with the label). This matches the `full-bleed` configuration nimble has today for the `nimble-text-field`. Therefore, the select, combobox, and number field will all be updated to also support being `full-bleed`.
+
+Similar to the `nimble-text-field`, these components will only support `full-bleed` when their appearance is `frameless`.
+
+## Separation of responsibility
+
+With the changes described above implemented in nimble, an application can do the following:
+
+- Create an Angular form with nimble controls configured with `readOnlyDisabledAppearance`
+- Update controls to use `frameless` appearance and `full-bleed` when they are disabled and will be styled as read-only
```

Comment:
@jattasNI and @rajsite - I'm curious if you think this is the correct separation of responsibility with the new direction to leave form controls as disabled but style them as readonly based on the presence of a new attribute. In that scenario, should nimble be more opinionated and style things as `frameless` + `full-bleed` regardless how these properties are configured on the component?

Reply:
I think conceptually we want `readonly` and `disabled[appearance-readonly]` to be as close as possible visually. Right now [`readonly` seems to respond to different appearance modes](https://nimble.ni.dev/storybook/index.html?path=/story/components-text-field--block-text-field&args=readonly:!true).

If we were to adopt the behavior described for `appearance-readonly` we should change it for `readonly` as well. Not necessarily a deal breaker just thinking out loud. I don't think I feel strongly one way or another (struggling to think off pros/cons of more flexibility on readonly states looking "more control like" vs strictly "more text like").

Reply:
I don't think we want to change the way `readonly` works. You can see some screenshots in the "Alternatives" section of the document of where we use `readonly` controls with non-frameless appearances. We shouldn't break those usages.

If we want to keep `appearance-readonly` aligned with `readonly` (which I think is a reasonable desire), then I don't think we should "force" `frameless` + `full-bleed` styles with `appearance-readonly`.